applications/FluidDynamicsApplication/custom_elements/fluid_adjoint_element.cpp

```diff
@@ -0,0 +1,717 @@
+    std::vector<VariableData const*>& rVariables) const
+{
+    rVariables.resize(1);
+    rVariables[0] = &ADJOINT_FLUID_VECTOR_3;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+void FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::ThisExtensions::GetAuxiliaryVariables(
+    std::vector<VariableData const*>& rVariables) const
+{
+    rVariables.resize(1);
+    rVariables[0] = &AUX_ADJOINT_FLUID_VECTOR_1;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::FluidAdjointElement(IndexType NewId)
+    : BaseType(NewId)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::FluidAdjointElement(
+    IndexType NewId,
+    GeometryType::Pointer pGeometry)
+    : BaseType(NewId, pGeometry)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::FluidAdjointElement(
+    IndexType NewId,
+    GeometryType::Pointer pGeometry,
+    PropertiesType::Pointer pProperties)
+    : BaseType(NewId, pGeometry, pProperties)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::~FluidAdjointElement()
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+Element::Pointer FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<FluidAdjointElement>(
+        NewId, Element::GetGeometry().Create(ThisNodes), pProperties);
+    KRATOS_CATCH("");
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+Element::Pointer FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<FluidAdjointElement>(NewId, pGeom, pProperties);
+    KRATOS_CATCH("");
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+Element::Pointer FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::Clone(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<FluidAdjointElement>(
+        NewId, Element::GetGeometry().Create(ThisNodes), Element::pGetProperties());
+    KRATOS_CATCH("");
+}
+
+template <unsigned int TDim, unsigned int TNumNodes, class TAdjointElementData>
+int FluidAdjointElement<TDim, TNumNodes, TAdjointElementData>::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    TAdjointElementData::Check(*this, rCurrentProcessInfo);
```

Comment:
We should call the base class check in here.

Reply:
done :)